File: 3396-valid-word/valid-word.java
class Solution {
    public boolean isValid(String word) {
        if (word.length() < 3)
            return false;

        int vowels = 0, consonants = 0;

        for (char ch : word.toCharArray()) {
            if (!Character.isLetterOrDigit(ch))
                return false;

            if (Character.isLetter(ch)) {
                char lower = Character.toLowerCase(ch);
                if (lower == 'a' || lower == 'e' || lower == 'i' || lower == 'o' || lower == 'u')
                    vowels++;
                else
                    consonants++;
            }
        }

        // have to finish the iterator to check the special character (case 572: "Ya$")
        return vowels > 0 && consonants > 0;
    }
}

File: 3396-valid-word/valid-word.py
class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False

        vowels = 0
        consonants = 0

        for ch in word:
            if not ch.isalnum():
                return False

            if ch.isalpha():
                if ch.lower() in "aeiou":
                    vowels += 1
                else:
                    consonants += 1

        return vowels > 0 and consonants > 0
